x-pack/plugins/stack_connectors/server/connector_types/openai/lib/utils.ts

```diff
@@ -76,18 +81,34 @@ export function getRequestWithStreamOption(
 export const getAxiosOptions = (
   provider: string,
   apiKey: string,
-  stream: boolean
-): { headers: Record<string, string>; responseType?: ResponseType } => {
+  stream: boolean,
+  config?: Config
+): { headers: Record<string, string>; httpsAgent?: https.Agent; responseType?: ResponseType } => {
   const responseType = stream ? { responseType: 'stream' as ResponseType } : {};
+
   switch (provider) {
     case OpenAiProviderType.OpenAi:
       return {
-        headers: { Authorization: `Bearer ${apiKey}`, ['content-type']: 'application/json' },
+        headers: { Authorization: `Bearer ${apiKey}`, 'Content-Type': 'application/json' },
         ...responseType,
       };
     case OpenAiProviderType.AzureAi:
       return {
-        headers: { ['api-key']: apiKey, ['content-type']: 'application/json' },
+        headers: { 'api-key': apiKey, 'Content-Type': 'application/json' },
+        ...responseType,
+      };
+    case OpenAiProviderType.PkiOpenAi:
+      if (!config?.certPath || !config?.keyPath) {
+        throw new Error('Certificate and key paths are required for PKI authentication');
+      }
+      const httpsAgent = new https.Agent({
+        cert: fs.readFileSync(config.certPath),
```

Comment:
I guess this isn't intended to run on cloud?  For Kibana config where we allow cert data, we allow them to be customized with files or the readable data.  

https://www.elastic.co/guide/en/kibana/current/alert-action-settings-kb.html#:~:text=xpack.actions.customHostSettings%5Bn%5D.ssl.certificateAuthoritiesData